lib/version_utils.pm

```diff
@@ -405,6 +406,24 @@ sub is_sle {
     return check_version($query, $version, qr/\d{2}(?:-sp\d)?/);
 }
 
+=head2 is_sled
+
+Check if distribution is SLED with optional filter for:
+Version: <=12-sp3 =12-sp1 >11-sp1 >=15 15+ (>=15 and 15+ are equivalent)
+=cut
+
+sub is_sled {
+    my $query = shift;
+    my $version = shift // get_var('VERSION');
+
+    # Check if it is SLE and specifically SLED
+    return 0 unless is_sle($query, $version);
```

Comment:
Is this about is_sled or SLE version ? You just copied is_sle description, that does have  nothing with `is_sled` you will not even compare the SLED version at the end.

Why complicate simple thing ? Twisting it twice.
Why not just return 1 if ...SLED...

Reply:
Reduced to just:
`=head2 is_sled

Check if distribution is SLED
=cut

sub is_sled {
    return 0 unless is_sle();
    return 0 unless check_var('SLE_PRODUCT', 'sled') || get_var('FLAVOR') =~ /-Desktop/i;
    return 1;
}`

Reply:
`return 0 unless is_sle();` makes this broken, SLED is also SLE so also SLES would pass here.

Reply:
@dzedro What would be a better way? sles passes, then SLE_PRODUCT check is done

Reply:
1. Check how return works
2. [Why not just return 1 if ...SLED...](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20845#discussion_r1891640788)

Reply:
@dzedro  Thanks for your input. The is_sle() check is included to account for the || get_var('FLAVOR') =~ /-Desktop/i part, ensuring cases are handled where FLAVOR might suggest a desktop distribution even if SLE_PRODUCT isn’t explicitly set to 'sled'. If it's absolutely guaranteed that all SLED systems will always have SLE_PRODUCT set to 'sled', we could streamline this further. However, I haven't seen confirmation of that assumption. 

While I can rewrite it as:
`return check_var('SLE_PRODUCT', 'sled') || (is_sle() && (get_var('FLAVOR') =~ /-Desktop/i));`

I can’t say this version is any more or less readable—it’s a matter of preference without a linter or style guide to dictate the "correct" approach. If you feel strongly about this, I’m happy to adopt a specific convention if there’s one we should adhere to.

Let me know if this works better, or if we’re confident SLE_PRODUCT is sufficient, I can simplify it even further.

Reply:
If SLE_PRODUCT is not set to SLED then it should if there is no -Desktop flavor
What forces you to use is_sle in function which is testing if the system is SLED ? There is no reason AFAIK.

Reply:
On SLED `SLE_PRODUCT` should be set to `sled`, done. nothing else is needed.

Reply:
> On SLED SLE_PRODUCT should be set to sled, done. Nothing else is needed.

Okay, your point is clear now. Then I am reducing code to just `return check_var('SLE_PRODUCT', 'sled')` and expecting variable `SLE_PRODUCT` to be set to `sled` for whatever `sled` oriented tests

Reply:
P.S.
The phrasing is a bit unclear to me. Could you clarify what you mean by it should be if there is no -Desktop flavor'? Are you suggesting that the logic should explicitly check for FLAVOR not containing -Desktop if SLE_PRODUCT is not set to sled?

Reply:
@dzedro I think the latest code correction adheres to your suggestion